Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.cpp

```diff
@@ -104,21 +126,170 @@ namespace EMStudio
+            AzFramework::ViewportColors::ZAxisColor);
+        m_rotateManipulators.InstallMouseMoveCallback(
+            [this](const AzToolsFramework::AngularManipulator::Action& action)
+            {
+                OnManipulatorRotated(action.LocalOrientation());
+            });
+
+        // Setup the scale manipulator
+        m_scaleManipulators.SetSpace(worldTransform);
+        m_scaleManipulators.SetAxes(AZ::Vector3::CreateAxisX(), AZ::Vector3::CreateAxisY(), AZ::Vector3::CreateAxisZ());
+        m_scaleManipulators.ConfigureView(
+            AzToolsFramework::LinearManipulatorAxisLength(), AzFramework::ViewportColors::XAxisColor,
+            AzFramework::ViewportColors::YAxisColor, AzFramework::ViewportColors::ZAxisColor);
+        m_scaleManipulators.InstallAxisMouseMoveCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                OnManipulatorScaled(action.LocalScale(), action.LocalScaleOffset());
+            });
+    }
+
+    void AtomRenderPlugin::SetManipulatorMode(RenderOptions::ManipulatorMode mode)
+    {
+        if (!m_manipulatorManager)
+        {
+            return;
+        }
+
+        if (mode == RenderOptions::ManipulatorMode::SELECT)
+        {
+            m_translationManipulators.Unregister();
+            m_rotateManipulators.Unregister();
+            m_scaleManipulators.Unregister();
+            return;
+        }
+
+        const AZ::EntityId entityId = m_animViewportWidget->GetAnimViewportRenderer()->GetEntityId();
+        if (!entityId.IsValid())
+        {
+            return;
+        }
+
+        AZ::Vector3 localPos;
+        AZ::TransformBus::EventResult(localPos, entityId, &AZ::TransformBus::Events::GetLocalTranslation);
+
+        switch (mode)
+        {
+        case RenderOptions::ManipulatorMode::SELECT:
+            // The AtomRenderPlugin doesn't implement a select mode
+            break;
+        case RenderOptions::ManipulatorMode::TRANSLATE:
+            {
+                m_translationManipulators.Register(g_animManipulatorManagerId);
+                m_translationManipulators.SetLocalPosition(localPos);
+                m_rotateManipulators.Unregister();
+                m_scaleManipulators.Unregister();
+            }
+            break;
+        case RenderOptions::ManipulatorMode::ROTATE:
+            {
+                m_translationManipulators.Unregister();
+                m_rotateManipulators.Register(g_animManipulatorManagerId);
+                m_rotateManipulators.SetLocalPosition(localPos);
+                m_scaleManipulators.Unregister();
+            }
+            break;
+        case RenderOptions::ManipulatorMode::SCALE:
+            {
+                m_translationManipulators.Unregister();
+                m_rotateManipulators.Unregister();
+                m_scaleManipulators.Register(g_animManipulatorManagerId);
+                m_scaleManipulators.SetLocalPosition(localPos);
```

Comment:
suggestion: There's probably a way to write this without so much repetition but it's fine for now, could be a nice simplification/refactoring in future